lib/mocha.js

```diff
@@ -95,29 +96,47 @@ function Mocha(options) {
   options = options || {};
   this.files = [];
   this.options = options;
-  if (options.grep) {
-    this.grep(new RegExp(options.grep));
-  }
-  if (options.fgrep) {
-    this.fgrep(options.fgrep);
-  }
+  // root suite
   this.suite = new exports.Suite('', new exports.Context());
-  this.ui(options.ui);
-  this.bail(options.bail);
-  this.reporter(options.reporter, options.reporterOptions);
-  if (typeof options.timeout !== 'undefined' && options.timeout !== null) {
-    this.timeout(options.timeout);
+
+  this.grep(options.grep)
+    .fgrep(options.fgrep)
```

Comment:
`Mocha#grep` and `Mocha#fgrep` are really an "either or" choice.
`options.fgrep` has priority.

Reply:
this is true, but probably undocumented and untested.

- [ ] make these conflict